fe1-web/model/network/method/message/data/MessageProperties.ts

```diff
@@ -0,0 +1,60 @@
+// eslint-disable-next-line max-classes-per-file
+import { ActionType, ObjectType } from './MessageData';
+
+/**
+ * Interface to describe the properties of a certain type of message.
+ *
+ * @remarks
+ * For now, there are only 2 classes of messages: the ones that are signed using a token, and the
+ * one that use the public key as a signature. (2021-12-27, Xelowak)
+ */
+export interface MessageProperties {
+  readonly isPopTokenSigned: boolean;
+}
+
+class PopTokenSignedMessage implements MessageProperties {
+  readonly isPopTokenSigned = true;
+}
+
+class PublicKeySignedMessage implements MessageProperties {
+  readonly isPopTokenSigned = false;
+}
+
+/**
+ * Map to know the properties of the type of message we're going to send, according to the object
+ * and the action.
+ *
+ * @remarks
+ * This map has to be updated for each new kind of message, and each new properties we would like
+ * to add.
+ */
+export const messagePropertiesMap: Map<ObjectType, Map<ActionType, MessageProperties>> = new Map([
+  [ObjectType.CHIRP, new Map<ActionType, MessageProperties>([
+    [ActionType.ADD, new PopTokenSignedMessage()],
```

Comment:
I'm not going to annoy you further, but this could be simply :

```suggestion
    [ActionType.ADD, { isPopTokenSigned: true }],
```

Reply:
Effectively this means you wouldn't need classes at all, just this gigantic table which we'll be able to expand in the future :)

Reply:
Thanks, I did it ! I think it's now ready to merge